src/Core/Commander/ManagerCommands.js

```diff
@@ -40,6 +40,9 @@ define('Core/Commander/ManagerCommands', [
             this.providerMap = {};
             this.history = null;
             this.eventsManager = new EventsManager();
+            this.currentCommandsCount = 0;
+            this.maxConcurrentCommands = 16;
```

Comment:
So, I don't think this mechanism is really suited to our case. If I understand your links correctly, the goal of the tick function is to execute the maximum of atoms in 16ms. However, we don't have a lot to do at each frame, we just create new promises. The only big operation is the tree browsing which can't (easily) be cut into multiple atoms.
However, it may be useful in the future, when multiple quadtrees (or other data structures) will need updating.